### PR TITLE
webpack fixes for metamask config

### PR DESCRIPTION
**Pull request by naugtur**
- rootDir setting to let aa find the right package.json while webpack `compiler.context` is set to a subpath
- prevent error in policy generation for virtual modules webpack uses for juggling things that don't actually have sources
- added some debugging helpers

### CODE REVIEW

File: packages/webpack/src/buildtime/policyGenerator.js [hunk 1]
```diff
@@ -99,6 +100,14 @@ module.exports = {
         // Skip modules the user intentionally excludes.
         // This is policy generation so we don't need to protect ourselves from an attack where the module has a loader defined in the specifier.
         if (isExcludedUnsafe(module)) return
+        if (module.userRequest === undefined) {
+          diag.rawDebug(
+            1,
+            `LavaMoatPlugin: Module ${module} has no userRequest`
+          )
+          diag.rawDebug(2, { skippingInspectingModule: module })
```
Comment (boneskull) [inline packages/webpack/src/buildtime/policyGenerator.js:108]: what is this option?

File: packages/webpack/src/plugin.js [hunk 2]
```diff
@@ -102,6 +102,18 @@ class LavaMoatPlugin {
     })
 
     const options = this.options
+
+    diag.run(2, () => {
```
Comment (boneskull) [inline packages/webpack/src/plugin.js:106]: what is this magic number?

File: packages/webpack/src/plugin.js [hunk 3]
```diff
@@ -102,6 +102,18 @@ class LavaMoatPlugin {
...
+      // Log stack traces for all errors on higher verbosity because webpack won't
+      compiler.hooks.done.tap(PLUGIN_NAME, (stats) => {
+        if (stats.hasErrors()) {
+          stats.compilation.errors.forEach((error) => {
+            console.error(error.stack || error)
```
Comment (boneskull) [inline packages/webpack/src/plugin.js]:
```suggestion
            console.error(error)
```

Maybe better to let Node.js decide how to display it?

**Review by boneskull — DISMISSED** [review]
I don't understand how any of this stuff works, but the changes seem fine!

File: packages/webpack/src/buildtime/policyGenerator.js (hunk 1 above)
Comment (naugtur) [reply packages/webpack/src/buildtime/policyGenerator.js:108]: It's just a console log based on the diagnostics level. The object is what gets logged

File: packages/webpack/src/plugin.js (hunk 2 above)
Comment (naugtur) [reply packages/webpack/src/plugin.js:106]: The log level

Comment (boneskull) [reply packages/webpack/src/plugin.js:106]:
```suggestion
    const LOG_LEVEL = 2
    diag.run(LOG_LEVEL, () => {
```

😄

**Review by weizman — APPROVED** [review]
lgtm

**Pull request merged**